drivers/i2c/i2c_ite_it8xxx2.c

```diff
@@ -746,7 +763,15 @@ static int i2c_it8xxx2_transfer(const struct device *dev, struct i2c_msg *msgs,
 		/* Start transaction */
 		i2c_transaction(dev);
 		/* Wait for the transfer to complete */
-		k_sem_take(&data->device_sync_sem, K_FOREVER);
+		/* TODO: the timeout should be adjustable */
+		res = k_sem_take(&data->device_sync_sem, K_MSEC(100));
+		if (res != 0) {
+			data->err = ETIMEDOUT;
+			/* reset i2c port */
+			i2c_reset(dev, I2C_RC_TIMEOUT);
+			/* If this message is sent fail, drop the transaction. */
+			return -EINVAL;
```

Comment:
Why -EINVAL? Should it not be -ETIMEDOUT ?

Why is there a separate data->err value? Is that because the errors happen in interrupts?

Reply:
There should be modified to "break;".  If k_sem_take is time out, it will break and then run the follow-up program and return ETIMEDOUT. 

data->err stores the error status of the transaction and it will be returned in this function.